pkgs/build-support/go/module.nix

```diff
@@ -58,34 +58,54 @@
 assert goPackagePath != "" -> throw "`goPackagePath` is not needed with `buildGoModule`";
 
 let
-  args = removeAttrs args' [ "overrideModAttrs" "vendorSha256" "vendorHash" ];
```

Comment:
This is a bit surprising, could you explain this removal?

Reply:
This is because we now pass `vendorHash` as a derivation attribute, so that the `configurePhase` of the main derivation can access `finalAttrs.vendorHash` as `"$vendorHash"`.

Indeed I could keep it in `passthru`, but it complicates the imementation.

People can access `pet.vendorHash` as before.